velox/functions/prestosql/aggregates/GeometricMeanAggregate.cpp

```diff
@@ -0,0 +1,141 @@
+      logSum_ += std::log(data);
+      count_ = checkedPlus<int64_t>(count_, 1);
+    }
+
+    void combine(
+        HashStringAllocator* /*allocator*/,
+        exec::arg_type<Row<double, int64_t>> other) {
+      // Use VELOX_USER_CHECK here to make aggregation fuzzer happy.
+      VELOX_USER_CHECK(other.at<0>().has_value());
+      VELOX_USER_CHECK(other.at<1>().has_value());
+      logSum_ += other.at<0>().value();
+      count_ = checkedPlus<int64_t>(count_, other.at<1>().value());
+    }
+
+    bool writeFinalResult(exec::out_type<OutputType>& out) {
+      out = std::exp(logSum_ / count_);
+      return true;
+    }
+
+    bool writeIntermediateResult(exec::out_type<IntermediateType>& out) {
+      out = std::make_tuple(logSum_, count_);
+      return true;
+    }
+  };
+};
+
+} // namespace
+
+void registerGeometricMeanAggregate(const std::string& prefix) {
+  const std::string name = prefix + kGeometricMean;
+
+  std::vector<std::shared_ptr<exec::AggregateFunctionSignature>> signatures;
+
+  for (const auto& inputType : {"bigint", "double"}) {
+    signatures.push_back(exec::AggregateFunctionSignatureBuilder()
+                             .returnType("double")
+                             .intermediateType("row(double,bigint)")
+                             .argumentType(inputType)
+                             .build());
+  }
+
+  exec::registerAggregateFunction(
+      name,
+      std::move(signatures),
+      [name](
+          core::AggregationNode::Step step,
+          const std::vector<TypePtr>& argTypes,
+          const TypePtr& resultType,
+          const core::QueryConfig& /*config*/)
+          -> std::unique_ptr<exec::Aggregate> {
+        VELOX_CHECK_EQ(argTypes.size(), 1, "{} takes one argument", name);
+        auto inputType = argTypes[0];
+        if (exec::isRawInput(step)) {
+          switch (inputType->kind()) {
+            case TypeKind::BIGINT:
+              return std::make_unique<
+                  SimpleAggregateAdapter<GeometricMeanAggregate<int64_t>>>(
+                  resultType);
+            case TypeKind::DOUBLE:
+              return std::make_unique<
+                  SimpleAggregateAdapter<GeometricMeanAggregate<double>>>(
+                  resultType);
+            default:
+              VELOX_USER_FAIL(
+                  "Unknown input type for {} aggregation {}",
+                  name,
+                  inputType->toString());
+          }
+        } else {
+          switch (resultType->kind()) {
+            case TypeKind::DOUBLE:
+            case TypeKind::ROW:
+              return std::make_unique<
+                  SimpleAggregateAdapter<GeometricMeanAggregate<double>>>(
+                  resultType);
+            default:
+              VELOX_USER_FAIL(
+                  "Unsupported result type for final aggregation: {}",
+                  resultType->toString());
+          }
+        }
+      },
+      true);
```

Comment:
change this to false to skip registering companion functions; this should help avoid fuzzer failures

Reply:
done